src/fermilib/utils/_operator_utils.py

```diff
@@ -151,19 +150,17 @@ def save_operator(operator, file_name=None, data_directory=None):
     else:
         raise TypeError('Operator of invalid type.')
 
-    with h5py.File(file_path, 'w') as f:
-        f['operator_type'] = numpy.string_(operator_type)
-        f['operator_terms'] = numpy.string_(json.dumps(
-            [{'k': k, 'r': v.real, 'i': v.imag}
-                for k, v in operator.terms.items()],
-            ensure_ascii=False).encode('utf8'))
+    tm = operator.terms
+    f = open(file_path, 'wb')
+    marshal.dump((operator_type, {k: numpy_scalar(tm[k]) for k in tm}), f)
```

Comment:
I am not sure we can trust that it will always be initialized this way. Even if it is, inevitably people will write things like:
my_fermion_operator *= my_numpy_array[i]
which may cause the coefficient to become a numpy type. So I think we are going to need to cast directly.